ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/transaction/SandboxTransactionService.scala

```diff
@@ -173,50 +178,6 @@ class SandboxTransactionService private (val ledgerBackend: LedgerBackend, paral
   override lazy val offsetOrdering: Ordering[LedgerOffset.Absolute] =
     Ordering.by(abs => BigInt(abs.value))
 
-  private def runTransactionPipeline(
-      requestingParties: List[Party],
-      begin: LedgerOffset,
-      end: Option[LedgerOffset]): Source[AcceptedTransaction, NotUsed] = {
-
-    Source.fromFuture(ledgerBackend.getCurrentLedgerEnd).flatMapConcat { ledgerEnd =>
-      OffsetSection(begin, end)(getOffsetHelper(ledgerEnd)) match {
-        case Failure(exception) => Source.failed(exception)
-        case Success(value) =>
-          value match {
-            case OffsetSection.Empty => Source.empty
-            case OffsetSection.NonEmpty(subscribeFrom, subscribeUntil) =>
-              ledgerBackend
-                .ledgerSyncEvents(Some(subscribeFrom))
-                .takeWhile(
-                  {
-                    case item =>
-                      // the offset we get from LedgerBackend is the actual offset of the entry. We need to return the next one
-                      // however on the API so clients can resubscribe with the received offset without getting duplicates
-                      subscribeUntil.fold(true)(until => until != (item.offset.toLong + 1).toString)
-                  },
-                  inclusive = true
-                )
-                .collect {
-                  case t: AcceptedTransaction => t.copy(offset = (t.offset.toLong + 1).toString)
-                } //again, returning the next offset one
-          }
-      }
-    }
-  }
-
-  private def getOffsetHelper(ledgerEnd: String) = {
-    new OffsetHelper[String] {
-      override def fromOpaque(opaque: String): Try[String] = Success(opaque)
-
-      override def getLedgerBeginning(): String = "0"
-
-      override def getLedgerEnd(): String = ledgerEnd
-
-      override def compare(o1: String, o2: String): Int =
-        java.lang.Long.compare(o1.toLong, o2.toLong)
-    }
-  }
-
   private def lookUpByTransactionId(
```

Comment:
Please also use `transactionPipeline` in this method, so the offset is handled here properly as well.

Reply:
good catch. That logic is ultra inefficient with Postgres btw. I will create an issue to address that.

Reply:
I changed it @gerolf-da. Please take a look.